swift/Samples/AlertContainer/Sources/AlertContainerViewController.swift

```diff
@@ -0,0 +1,258 @@
+import UIKit
+import Workflow
+import WorkflowUI
+
+internal final class AlertContainerViewController<AlertScreen: Screen>: ScreenViewController<AlertContainerScreen<AlertScreen>> {
+    private var baseScreenViewController: DescribedViewController
+    
+    private let dimmingView = UIView()
+    
+    private var alertView: AlertView?
+    
+    required init(screen: AlertContainerScreen<AlertScreen>, environment: ViewEnvironment) {
+        baseScreenViewController = DescribedViewController(screen: screen.baseScreen, environment: environment)
+        super.init(screen: screen, environment: environment)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        addChild(baseScreenViewController)
+        view.addSubview(baseScreenViewController.view)
+        baseScreenViewController.didMove(toParent: self)
+        
+        dimmingView.backgroundColor = UIColor(white: 0, alpha: 0.5)
+        view.addSubview(dimmingView)
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        
+        baseScreenViewController.view.frame = view.bounds
+        
+        dimmingView.frame = view.bounds
+        dimmingView.isUserInteractionEnabled = (alertView != nil)
+        dimmingView.alpha = (alertView != nil) ? 1 : 0
+    }
+    
+    override func screenDidChange(from previousScreen: AlertContainerScreen<AlertScreen>, previousEnvironment: ViewEnvironment) {
+        update()
```

Comment:
does this need to call `super.screenDidChange`? (asking out of ignorance as I actually don't know 😄)

Reply:
Its a good question. The default implementation is empty and has a comment in ScreenViewController says `Subclasses should override this method in order to update any relevant UI bits when the screen model changes.` I don't see any other class calling super inside this function so i followed the same convention. We will likely need changes if we end up adding any default implementation. I don't have a strong opinion either way.

Reply:
In this case, `super.screen...` doesn't make a difference. However, it's best practice to always call it - since the base class might add functionality to it's implementation at some-point (or we may change the base class and inherit from another which does implement it).

Reply:
i've added `super.screenDidChange` call.